cx_Freeze fails to install in platforms not supported by patchelf-pypi

Help me:
--------------------------------------------------------------------------------
      -- Trying "Ninja" generator
      --------------------------------
      ---------------------------
      ----------------------
      -----------------
      ------------
      -------
      --
      Not searching for unused variables given on the command line.
      CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
      -- Configuring incomplete, errors occurred!
      See also "/data/data/com.termux/files/usr/tmp/pip-install-bedqnhtz/patchelf_3243bbffba8f4fd787e8ce6140e21522/_cmake_test_compile/build/CMakeFiles/CMakeOutput.log".
      --
      -------
      ------------
      -----------------
      ----------------------
      ---------------------------
      --------------------------------
      -- Trying "Ninja" generator - failure
      --------------------------------------------------------------------------------



      --------------------------------------------------------------------------------
      -- Trying "Unix Makefiles" generator
      --------------------------------
      ---------------------------
      ----------------------
      -----------------
      ------------
      -------
      --
      Not searching for unused variables given on the command line.
      -- The C compiler identification is Clang 15.0.6
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /data/data/com.termux/files/usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- The CXX compiler identification is Clang 15.0.6
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /data/data/com.termux/files/usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Configuring done
      -- Generating done
      -- Build files have been written to: /data/data/com.termux/files/usr/tmp/pip-install-bedqnhtz/patchelf_3243bbffba8f4fd787e8ce6140e21522/_cmake_test_compile/build
      --
      -------
      ------------
      -----------------
      ----------------------
      ---------------------------
      --------------------------------
      -- Trying "Unix Makefiles" generator - success
      --------------------------------------------------------------------------------

      Configuring Project
        Working directory:
          /data/data/com.termux/files/usr/tmp/pip-install-bedqnhtz/patchelf_3243bbffba8f4fd787e8ce6140e21522/_skbuild/linux-armv7l-3.11/cmake-build
        Command:
          cmake /data/data/com.termux/files/usr/tmp/pip-install-bedqnhtz/patchelf_3243bbffba8f4fd787e8ce6140e21522 -G 'Unix Makefiles' -DCMAKE_INSTALL_PREFIX:PATH=/data/data/com.termux/files/usr/tmp/pip-install-bedqnhtz/patchelf_3243bbffba8f4fd787e8ce6140e21522/_skbuild/linux-armv7l-3.11/cmake-install -DPYTHON_VERSION_STRING:STRING=3.11.1 -DSKBUILD:INTERNAL=TRUE -DCMAKE_MODULE_PATH:PATH=/data/data/com.termux/files/usr/tmp/pip-build-env-frruy6lx/overlay/lib/python3.11/site-packages/skbuild/resources/cmake -DPYTHON_EXECUTABLE:PATH=/data/data/com.termux/files/home/.local/share/virtualenvs/test-GelrdIJB/bin/python -DPYTHON_INCLUDE_DIR:PATH=/data/data/com.termux/files/usr/include/python3.11 -DPYTHON_LIBRARY:PATH=/data/data/com.termux/files/usr/lib/libpython3.11.so -DPython_EXECUTABLE:PATH=/data/data/com.termux/files/home/.local/share/virtualenvs/test-GelrdIJB/bin/python -DPython_ROOT_DIR:PATH=/data/data/com.termux/files/home/.local/share/virtualenvs/test-GelrdIJB -DPython_INCLUDE_DIR:PATH=/data/data/com.termux/files/usr/include/python3.11 -DPython_FIND_REGISTRY:STRING=NEVER -DPython3_EXECUTABLE:PATH=/data/data/com.termux/files/home/.local/share/virtualenvs/test-GelrdIJB/bin/python -DPython3_ROOT_DIR:PATH=/data/data/com.termux/files/home/.local/share/virtualenvs/test-GelrdIJB -DPython3_INCLUDE_DIR:PATH=/data/data/com.termux/files/usr/include/python3.11 -DPython3_FIND_REGISTRY:STRING=NEVER -DCMAKE_BUILD_TYPE:STRING=Release

      -- The C compiler identification is Clang 15.0.6
      -- The CXX compiler identification is Clang 15.0.6
      -- Detecting C compiler ABI info
      -- Detecting C compiler ABI info - done
      -- Check for working C compiler: /data/data/com.termux/files/usr/bin/cc - skipped
      -- Detecting C compile features
      -- Detecting C compile features - done
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /data/data/com.termux/files/usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      -- Configuring done
      -- Generating done
      CMake Warning:
        Manually-specified variables were not used by the project:

          PYTHON_EXECUTABLE
          PYTHON_INCLUDE_DIR
          PYTHON_LIBRARY
          PYTHON_VERSION_STRING
          Python3_EXECUTABLE
          Python3_FIND_REGISTRY
          Python3_INCLUDE_DIR
          Python3_ROOT_DIR
          Python_EXECUTABLE
          Python_FIND_REGISTRY
          Python_INCLUDE_DIR
          Python_ROOT_DIR
          SKBUILD


      -- Build files have been written to: /data/data/com.termux/files/usr/tmp/pip-install-bedqnhtz/patchelf_3243bbffba8f4fd787e8ce6140e21522/_skbuild/linux-armv7l-3.11/cmake-build
      [ 12%] Creating directories for 'build_patchelf'
      [ 25%] No download step for 'build_patchelf'
      [ 37%] No update step for 'build_patchelf'
      [ 50%] Performing patch step for 'build_patchelf'
      ./bootstrap.sh: 2: autoreconf: not found
      make[2]: *** [CMakeFiles/build_patchelf.dir/build.make:113: build_patchelf-prefix/src/build_patchelf-stamp/build_patchelf-patch] Error 127
      make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/build_patchelf.dir/all] Error 2
      make: *** [Makefile:136: all] Error 2
      Traceback (most recent call last):
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-frruy6lx/overlay/lib/python3.11/site-packages/skbuild/setuptools_wrap.py", line 640, in setup
          cmkr.make(make_args, install_target=cmake_install_target, env=env)
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-frruy6lx/overlay/lib/python3.11/site-packages/skbuild/cmaker.py", line 684, in make
          self.make_impl(clargs=clargs, config=config, source_dir=source_dir, install_target=install_target, env=env)
        File "/data/data/com.termux/files/usr/tmp/pip-build-env-frruy6lx/overlay/lib/python3.11/site-packages/skbuild/cmaker.py", line 715, in make_impl
          raise SKBuildError(

      An error occurred while building with CMake.
        Command:
          cmake --build . --target install --config Release --
        Install target:
          install
        Source directory:
          /data/data/com.termux/files/usr/tmp/pip-install-bedqnhtz/patchelf_3243bbffba8f4fd787e8ce6140e21522
        Working directory:
          /data/data/com.termux/files/usr/tmp/pip-install-bedqnhtz/patchelf_3243bbffba8f4fd787e8ce6140e21522/_skbuild/linux-armv7l-3.11/cmake-build
      Please check the install target is valid and see CMake's output for more information.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for patchelf
Successfully built cx-freeze
Failed to build patchelf
ERROR: Could not build wheels for patchelf, which is required to install pyproject.toml-based projects

--- a/cx_Freeze/parser.py
+++ b/cx_Freeze/parser.py
@@ -273,8 +273,8 @@ def _verify_patchelf() -> None:
         raise ValueError("Could not call `patchelf` binary") from None
 
     mobj = re.match(r"patchelf\s+(\d+(.\d+)?)", version)
-    if mobj and tuple(int(x) for x in mobj.group(1).split(".")) >= (0, 12):
+    if mobj and tuple(int(x) for x in mobj.group(1).split(".")) >= (0, 14):
         return
     raise ValueError(
-        f"patchelf {version} found. cx_Freeze requires patchelf >= 0.12."
+        f"patchelf {version} found. cx_Freeze requires patchelf >= 0.14."
     )

--- a/doc/src/faq.rst
+++ b/doc/src/faq.rst
@@ -26,9 +26,9 @@ packages, this may lead to undesired behavior, such as **cx_Freeze**
 encountering a recursion error when trying to compute the list of dependencies,
 or the `lib` folder of the frozen application containing many unnecessary
 packages.
-In this case, you may want to use **cx_Freeze** in a virtualenv. Alternatively, the
-`setup_script` also offers the :option:`excludes` option to explicitly exclude
-dependencies which would otherwise be included.
+In this case, you may want to use **cx_Freeze** in a virtualenv. Alternatively,
+the `setup_script` also offers the :option:`excludes` option to explicitly
+exclude dependencies which would otherwise be included.
 
 Specifying modules and packages
 -------------------------------
@@ -163,3 +163,37 @@ included with the frozen program.
 In order to make it easy to comply with this requirement, **cx_Freeze** will
 automatically include a copy of the license, as a text file, as part of
 the frozen program.
+
+.. _patchelf:
+
+How to install Patchelf
+-----------------------
+
+Patchelf is used in Linux and unix-like systems (FreeBSD, etc, except macOS).
+In Linux, cx_Freeze 6.10+ installs it using |PyPI_link_patchelf| wheels.
+
+ .. |PyPI_link_patchelf| raw:: html
+
+   <a href="https://pypi.org/project/patchelf/" target="_blank">PyPI</a>
+
+If you have any trouble with it, because your platform is not supported by
+binary wheels, please install it using the system package manager or from
+sources.
+
+ To install patchelf in debian-based:
+
+  .. code-block:: console
+
+    sudo apt-get install patchelf
+
+ To install patchelf in fedora:
+
+  .. code-block:: console
+
+    dnf install patchelf
+
+ Or install patchelf from |patchelf_sources|.
+
+ .. |patchelf_sources| raw:: html
+
+   <a href="https://github.com/NixOS/patchelf#compiling-and-testing" target="_blank">sources</a>

--- a/doc/src/installation.rst
+++ b/doc/src/installation.rst
@@ -31,43 +31,17 @@ Python requirements are installed automatically by pip or conda.
    cx_Logging >= 3.1           (Windows only)
    lief >= 0.11.5              (Windows only)
    packaging >= 21.0
-   patchelf >= 0.12            (Linux)
+   patchelf >= 0.14            (Linux)
    setuptools >= 59.0.1
 
 .. note:: Patchelf
 
- patchelf is used in Linux and unix-like systems (FreeBSD, etc), except macOS.
- In Linux, cx_Freeze 6.10+ installs it using new wheels available on
- |PyPI_link_patchelf|.
-
- .. |PyPI_link_patchelf| raw:: html
-
-   <a href="https://pypi.org/project/patchelf/" target="_blank">PyPI</a>
-
- If you have any trouble with it, use the old method:
-
- To install patchelf in debian/ubuntu:
-
-  .. code-block:: console
-
-    sudo apt-get install patchelf
-
- To install patchelf in fedora:
-
-  .. code-block:: console
-
-    dnf install patchelf
-
- Or install patchelf from |patchelf_sources|
-
- .. |patchelf_sources| raw:: html
-
-   <a href="https://github.com/NixOS/patchelf#compiling-and-testing" target="_blank">sources</a>
+ If you have any trouble with patchelf, check :ref:`patchelf`.
 
 Pipenv
 ------
 
-Using pipenv, install or update by issuing one of the folowing commanda:
+Using pipenv, install or update by issuing one of the folowing command:
 
   .. code-block:: console
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -40,7 +40,11 @@ dependencies = [
     "setuptools>=61.2,<66",
     "cx_Logging>=3.1 ;sys_platform == 'win32'",
     "lief>=0.12.0 ;sys_platform == 'win32'",
-    "patchelf>=0.13 ;sys_platform == 'linux'",
+    "patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'x86_64'",
+    "patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'i686'",
+    "patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'aarch64'",
+    "patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'ppc64le'",
+    "patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 's390x'",
 ]
 
 [project.readme]

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -4,7 +4,11 @@ packaging>=21.3
 setuptools>=61.2,<66
 cx_Logging>=3.1 ;sys_platform == 'win32'
 lief>=0.12.0 ;sys_platform == 'win32'
-patchelf>=0.13 ;sys_platform == 'linux'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'x86_64'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'i686'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'aarch64'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'ppc64le'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 's390x'
 
 # dev
 bump2version>=1.0.1

--- a/requirements.txt
+++ b/requirements.txt
@@ -4,4 +4,8 @@ packaging>=21.3
 setuptools>=61.2,<66
 cx_Logging>=3.1 ;sys_platform == 'win32'
 lief>=0.12.0 ;sys_platform == 'win32'
-patchelf>=0.13 ;sys_platform == 'linux'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'x86_64'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'i686'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'aarch64'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 'ppc64le'
+patchelf>=0.14 ;sys_platform == 'linux' and platform_machine == 's390x'

